### ISSUE TITLE
lockfile is deprecated

### ISSUE THREAD

**Issue by tuxdna** [projectatomic/atomicapp#599]
Sub Tasks:

 - [ ] - re-evaluate locking in general
 - [ ] - is it required for us?
 - [ ] - there are numerous other issues opened around locking in Atomic App so if it doesn't make sense we should get rid of it
 - [ ] - move to fasteners in the future?

We are using `lockfile` package for file locking. This has been marked as deprecated on upstream:

 * https://pypi.python.org/pypi/lockfile

> Note: This package is **deprecated**. It is highly preferred that instead of
using this code base that instead `fasteners`_ or `oslo.concurrency`_ is
used instead. For any questions or comments or further help needed
please email `openstack-dev`_ and prefix your email subject
with ``[oslo][pylockfile]`` (for a faster response).

We need to evaluate other alternatives mentioned in the comment above:

 * [oslo.concurrency](http://docs.openstack.org/developer/oslo.concurrency/usage.html)
 * [fasteners](http://fasteners.readthedocs.org/en/latest/)

**Comment by dustymabe** [comment]
@tuxdna that is news to me. We have two options:

- re-evaluate locking in general 
    * is it required for us?
    * there are numerous other issues opened around locking in Atomic App so if it doesn't make sense we should get rid of it
- move to fasteners in the future

If we move to fasteners we need to make sure there are corresponding packages in most of the distributions that we build on top of (CentOS, Fedora, Debian, etc).

**Comment by tuxdna** [comment]
@rtnpro  @dustymabe @cdrage : I found the requirement of lockfiles  for atomicaapp here

  - **prevent atomicapp from running the same image twice at the same time** https://github.com/projectatomic/atomicapp/issues/119

However what should happen if someone does:

```
$ atomicapp run myatomicapp
$ atomicapp run myatomicapp
```

i.e. consecutive runs of atomicapp. In this case lockfile would be of no use because this is what happens:

```
(env2)[saleem@host atomicapp]$ atomicapp run projectatomic/apache-centos7-atomicapp 
2016-03-15 17:27:26,349 - atomicapp.install - INFO - App name is projectatomic/apache-centos7-atomicapp, will be populated to /home/saleem/cdk/atomicapp
2016-03-15 17:27:26,350 - atomicapp.install - INFO - Component data exist in /home/saleem/cdk/atomicapp, skipping population...
2016-03-15 17:27:26,354 - atomicapp.nulecule_base - INFO - Artifacts for apache-centos7-atomicapp present for these providers: docker, marathon, kubernetes
2016-03-15 17:27:26,354 - atomicapp.install - INFO - Installing dependencies for apache-centos7-atomicapp
2016-03-15 17:27:26,359 - atomicapp.nulecule_base - INFO - Artifacts for apache-centos7-atomicapp present for these providers: docker, marathon, kubernetes
2016-03-15 17:27:26,359 - atomicapp.run - INFO - Using provider docker for component apache-centos7-atomicapp
046d30bf449521528068037e55492314020d9679179022d71d9c43a4eb8c1509
Usage of loopback devices is strongly discouraged for production use. Either use `--storage-opt dm.thinpooldev` or use `--storage-opt dm.no_warn_on_loop_devices=true` to suppress this warning.

(env2)[saleem@host atomicapp]$ docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS                NAMES
046d30bf4495        centos/httpd        "/run-httpd.sh"     4 seconds ago       Up 3 seconds        0.0.0.0:80->80/tcp   sharp_varahamihira

(env2)[saleem@host atomicapp]$ atomicapp run projectatomic/apache-centos7-atomicapp 
2016-03-15 17:27:34,473 - atomicapp.install - INFO - App name is projectatomic/apache-centos7-atomicapp, will be populated to /home/saleem/cdk/atomicapp
2016-03-15 17:27:34,474 - atomicapp.install - INFO - Component data exist in /home/saleem/cdk/atomicapp, skipping population...
2016-03-15 17:27:34,478 - atomicapp.nulecule_base - INFO - Artifacts for apache-centos7-atomicapp present for these providers: docker, marathon, kubernetes
2016-03-15 17:27:34,478 - atomicapp.install - INFO - Installing dependencies for apache-centos7-atomicapp
2016-03-15 17:27:34,483 - atomicapp.nulecule_base - INFO - Artifacts for apache-centos7-atomicapp present for these providers: docker, marathon, kubernetes
2016-03-15 17:27:34,483 - atomicapp.run - INFO - Using provider docker for component apache-centos7-atomicapp
daded2428c35f213f25d20dc5b209f7a40a53ae279a8974fe2913c8b93f1065c
Error response from daemon: Cannot start container daded2428c35f213f25d20dc5b209f7a40a53ae279a8974fe2913c8b93f1065c: failed to create endpoint kickass_mestorf on network bridge: Bind for 0.0.0.0:80 failed: port is already allocated
2016-03-15 17:27:35,270 - atomicapp.cli.main - ERROR - Exception caught: CalledProcessError()
2016-03-15 17:27:35,271 - atomicapp.cli.main - ERROR - Run the command again with -v option to get more information.
(env2)[saleem@host atomicapp]$ 
```

Second Apache instance in container fails to start because the port is already being used. I think it is the application that should take care of this situation.

I think adding locks is increasing the complexity of `atomicapp`. 

Do we really need locks? Any comments ?

**Comment by cdrage** [comment]
Closed via #704
